### ISSUE TITLE
LSP support?

### ISSUE THREAD

**Issue by VarLad** [rhysd/kiro-editor#17]
There's now https://lib.rs/crates/lsp-types by the way
Should make the integration easier

**Comment by rhysd** [comment]
Yes, I know the package and LSP is included in [future works](https://github.com/rhysd/kiro-editor#future-works).

**Comment by VarLad** [comment]
> Completion, go to definition and look up using language servers

Might I add, LSP has support for semantic highlighting as well as auto formatting. I hope you could look into that (since several languages are already supporting it)
